desktop-exporter/server.go

```diff
@@ -13,51 +14,92 @@ type Server struct {
 	traceStore *TraceStore
 }
 
-func getTracesHandler(traceStore *TraceStore) func(http.ResponseWriter, *http.Request) {
+func newTraceSummary(traceID string, spans []SpanData) (*TraceSummary, error) {
+	durationMS, err := getTraceDuration(spans)
+	if err != nil {
+		return nil, err
+	}
+
+	return &TraceSummary{
+		TraceID:    traceID,
+		SpanCount:  uint32(len(spans)),
+		DurationMS: durationMS,
+	}, nil
+}
+
+func newTraceData(store *TraceStore, traceID string) (*TraceData, error) {
```

Comment:
This isn't instantiating a `TraceData` so much as `get`ing or `fetch`ing it from the `TraceStore`, so let's name it that way

```suggestion
func getTraceData(store *TraceStore, traceID string) (*TraceData, error) {
```